src/shogun/preprocessor/RandomFourierGaussPreproc.cpp

```diff
@@ -140,264 +35,133 @@ EFeatureType RandomFourierGaussPreproc::get_feature_type() {
 }
 
 int32_t RandomFourierGaussPreproc::get_dim_feature_space() const {
-	return ((int32_t) dim_feature_space);
+	return m_dims;
 }
 
-void RandomFourierGaussPreproc::set_dim_feature_space(const int32_t dim) {
-	if (dim <= 0) {
-		throw ShogunException(
-				"void CRandomFourierGaussPreproc::set_dim_feature_space(const int32 dim): dim<=0 is not allowed");
-	}
-
-	dim_feature_space = dim;
-
-}
-
-int32_t RandomFourierGaussPreproc::get_dim_input_space() const {
-	return ((int32_t) dim_input_space);
+void RandomFourierGaussPreproc::set_dim_feature_space(int32_t dim)
+{
+	require(m_dims > 0, "Feature space dimension must be greater than zero");
+	m_dims = dim;
 }
 
-void RandomFourierGaussPreproc::set_kernelwidth(const float64_t kernelwidth2 ) {
-	if (kernelwidth2 <= 0) {
-		throw ShogunException(
-				"void CRandomFourierGaussPreproc::set_kernelwidth(const float64_t kernelwidth2 ): kernelwidth2 <= 0 is not allowed");
-	}
-	kernelwidth=kernelwidth2;
+void RandomFourierGaussPreproc::set_kernelwidth(float64_t kernelwidth ) 
+{
+	require(m_kernelwidth > 0, "Kernelwidth must be greater than zero");
+	m_kernelwidth = kernelwidth;
 }
 
-float64_t RandomFourierGaussPreproc::get_kernelwidth( ) const {
-	return (kernelwidth);
+float64_t RandomFourierGaussPreproc::get_kernelwidth() const {
+	return m_kernelwidth;
 }
 
-void RandomFourierGaussPreproc::set_dim_input_space(const int32_t dim) {
-	if (dim <= 0) {
-		throw ShogunException(
-				"void CRandomFourierGaussPreproc::set_dim_input_space(const int32 dim): dim<=0 is not allowed");
-	}
-
-	dim_input_space = dim;
-
-}
-
-bool RandomFourierGaussPreproc::test_rfinited() const {
-
-	if ((dim_feature_space ==  cur_dim_feature_space)
-			&& (dim_input_space > 0) && (dim_feature_space > 0)) {
-		if ((dim_input_space == cur_dim_input_space)&&(Math::abs(kernelwidth-cur_kernelwidth)<1e-5)) {
+void RandomFourierGaussPreproc::init()
+{
+	m_kernelwidth = 1;
```

Comment:
could you do this initialization in the declaration? Then we dont need the method